marktplaats/models/listing_image.py

```diff
@@ -21,3 +31,31 @@ def parse(cls, data):
                 image_data["extraExtraLargeUrl"],
             ))
         return images
+
+
+def fetch_listing_images(listing_id: str) -> list[str]:
+    """
+    Returns a list of image URLs for a given listing.
+    It scrapes the listing page and parses the ld+json objects on that page.
+    :param listing_id: The listing ID to get images for.
+    :return: A list of image URLs (https).
+    """
+    r = requests.get(
+        f"https://link.marktplaats.nl/{listing_id}",
+        headers=REQUEST_HEADERS)
+    r.raise_for_status()  # raises so we can stop the fetching on a higher level
+
+    soup = BeautifulSoup(r.text, 'html.parser')
+
+    images = []
+
+    # get the data objects from the HTML response
+    for data in soup.select('script[type="application/ld+json"]'):
+        parsed = json.loads(data.text)
+        # the list of image URLs is hidden within the product object
+        if type(parsed) is dict and parsed["@type"] == "Product":
+            for image in parsed["image"]:
+                # the returned images are in a format that don't include a scheme, so we add one manually
+                images.append(f"https:{image}")
```

Comment:
Because there can only be a single `"Product"`
```suggestion
        if type(parsed) is dict and parsed["@type"] == "Product":
            for image in parsed["image"]:
                # the returned images are in a format that don't include a scheme, so we add one manually
                images.append(f"https:{image}")
            break
```
You could also do some more complicated `filter` and `next` stuff to make it a bit more pythonic I guess, but this is fine